refactor: use pnpm, vite and vitest

stefanvanherwijnen opened the pull request:
instead of yarn, rollup and jest.

@perarnborg 
Please have a look at 
https://github.com/stefanvanherwijnen/lionel-oauth-client/blob/f4a05ae375fdb9f5b5c0633e4be6db36bffcb409/test/unit/accessToken.test.ts#L97

This now makes the test fail and is related to the expiration date of the token. Although I have no clue why it seemed to work before.

Also, this now only creates an ESM and UMD build. And I have not tested the pubishing process.

stefanvanherwijnen committed:
refactor: replace yarn with pnpm and jest with vitest

stefanvanherwijnen committed:
fix: make workflows use pnpm

stefanvanherwijnen committed:
fix: remove packageManager from package.json

stefanvanherwijnen committed:
chore: cleanup

stefanvanherwijnen committed:
fix: use vitest run

perarnborg commented:
@stefanvanherwijnen Thank you for contributing with your time on this!

To me all of these changes look good, and the tests run fine, even if take back the line https://github.com/stefanvanherwijnen/lionel-oauth-client/blob/f4a05ae375fdb9f5b5c0633e4be6db36bffcb409/test/unit/accessToken.test.ts#L97

It is strange to me why this line causes problems for you, since for instance the same check is made in the test `should get an access token if there is one in storage` – it is essentilly a check that the function getAccessToken works if the browser storage contains a token.

While these changes look good we need to evaluate them some more before we can merge this. We have started working on converting the lionel client codebase into a monorepo and in that process we have looked over the build pipeline as well. So we will get back to you when we have had the band width to do that!

We are probably ready to move away from Yarn berry at any rate, even though I have liked that projects ambitions.

stefanvanherwijnen committed:
chore: remove test-results

stefanvanherwijnen committed:
chore: update dependencies

stefanvanherwijnen committed:
test: readd test which now seems to be working fine

stefanvanherwijnen commented:
Thanks, it made no sense to me either why it wouldn't work, and it seems it now does work fine :confused: .
I readded the line, removed the test-results and updated the dependencies.

For some reason I can never get Yarn to work properly so I only use pnpm now. I released a lionel-oauth-client under my own scope for which I made these changes to get it to work, but if you're free to stick with Yarn.

I would recommended the switch to Vite (instead of Rollup) though (which at the moment is the same until Vite starts using Rolldown).
